### PR TITLE
Increase coverage for hyperbolic, exponentials, and complexes

### PR DESCRIPTION
**Pull request by oscargus**
#### Other comments
The removed code is checked later anyway.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/functions/elementary/tests/test_exponential.py [hunk 1]
```diff
@@ -138,13 +143,19 @@ def test_exp_taylor_term():
     assert exp(x).taylor_term(1, x) == x
     assert exp(x).taylor_term(3, x) == x**3/6
     assert exp(x).taylor_term(4, x) == x**4/24
+    assert exp(x).taylor_term(-1, x) == S.Zero
 
 
 def test_exp_MatrixSymbol():
     A = MatrixSymbol("A", 2, 2)
     assert exp(A).has(exp)
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/tests/test_exponential.py]: Maybe this test should use unchanged from sympy.core.expr

**Comment by oscarbenjamin** [comment]
Looks good

**Comment by oscarbenjamin** [comment]
Ironically codecov says this will:

> **decrease** coverage by `7.852%`.

File: sympy/functions/elementary/tests/test_exponential.py (hunk 1 above)
Comment (oscargus) [reply sympy/functions/elementary/tests/test_exponential.py]: I created #16997 instead... Although this is not covered there.

**Comment by oscargus** [comment]
Yes, not sure why. One of the test runs must have failed to upload the data. Still, the files under consideration has increased coverage (or at least some)...

**Pull request merged**